### PR TITLE
Deprecate edgeConfigId in favor of datastreamId

### PR DESCRIPTION
**Pull request by carterworks**
## Description

As mentioned in #995, we should rename `edgeConfigId` to `datastreamId`. This adds the `deprecated()` validator (as created in #1000, so merge this PR after that one) to `edgeConfigId`. It adds a warning whenever `edgeConfigId` is used in the `configure` command.

I did not use this validator on `datasetId` because it is deprecated in favor of a deeply nested value, nor did I deprecate `eventMergeId` because it does not have a replacement.

## Related Issue

https://jira.corp.adobe.com/browse/PDCL-10587

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (non-breaking change which does not add functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [x] I have made any necessary test changes and all tests pass.
- [x] I have run the Sandbox successfully.

### CODE REVIEW

**Review by jonsnyder — CHANGES_REQUESTED** [review]
This looks great. Can you update the functional tests so that they use datastreamId, and then add a test to check that the deprecated message is logged when using edgeConfigId, and it uses that as the datastreamId?

File: test/functional/specs/Command Logic/C11634155.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+test("Test C11634155: Deprecates options like edgeConfigId and warns with use", async t => {
+  const alloy = createAlloyProxy();
+  await t.debug();
+  const warning = await alloy.configureErrorMessage(config);
+
+  await t.expect(warning).match(/The field 'edgeConfigId' is deprecated./);
+});
+
+test.skip("Test C11634155: When specifying a deprecated option like edgeConfigId, it uses the specified alternative, datastreamId", async t => {
```
Comment (carterworks) [inline test/functional/specs/Command Logic/C11634155.js:47]: See https://github.com/adobe/alloy/pull/1000/files#r1227217825

File: src/core/buildAndValidateConfig.js [hunk 2]
```diff
@@ -57,6 +44,23 @@ const buildAllOnInstanceConfiguredExtraParams = (
   );
 };
 
+const wrapLoggerInQueue = logger => {
```
Comment (jonsnyder) [inline src/core/buildAndValidateConfig.js:47]: This wrapping doesn't correctly handle the "enabled" flag on the logger. This is fine, lets just add a comment to the logger to explain this.

**Review by jonsnyder — APPROVED** [review]

**Review by shammowla — APPROVED** [review]

File: test/functional/specs/Command Logic/C11634155.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+  const alloy = createAlloyProxy();
+  await alloy.configure(config);
+
+  await logger.warn.expectMessageMatching(
+    /The field 'edgeConfigId' is deprecated./
+  );
+});
+
+test("Test C11634155: When specifying a deprecated option like edgeConfigId, it uses the specified alternative, datastreamId", async t => {
```
Comment (ninaceban) [inline test/functional/specs/Command Logic/C11634155.js:51]: can we add a test where both configs are specified and assert that we throw an error?

Comment (carterworks) [reply test/functional/specs/Command Logic/C11634155.js:63]: Done.

**Comment by carterworks** [comment]
I will merge this after #1000 is merged.

**Pull request merged**